scripts/sbatch.py

```diff
@@ -0,0 +1,288 @@
+                if user_dict.get(username) is not None:
+                    return (
+                        user_dict[username]["allocation"] - user_dict[username]["used"]
+                    )
+                else:
+                    sys.exit(
+                        "No core hours usage info for {} {} from json file {}".format(
+                            account, username, json_file
+                        )
+                    )
+        else:
+            sys.exit(
+                "No core hours usage info for {} from json file {}".format(
+                    account, json_file
+                )
+            )
+
+
+def compare_hours_and_sbatch(requested_hours, available_hours, hpc, sl_to_sbatch):
+    """
+       Compare requested hour with available hours
+       and either submit the job or reject
+    """
+    if requested_hours <= available_hours:
+        cmd = "{} -M {} {}".format(SBATCH_BIN, hpc, sl_to_sbatch)
+        subprocess.call(cmd, shell=True)
+    else:
+        sys.exit("Not enought core hours left, please contact Jonney")
+
+
+def process_slurm_header(sl_file):
+    """Read a slurm file and returns the wanted header values as a dictionary"""
+    with open(sl_file) as f:
+        lines = f.readlines()
+    # SBATCH --account=nesi00213
+    header_dict = {h.value: None for h in SlurmHeader}
+    for header in header_dict.keys():
+        for line in lines:
+            if "SBATCH" in line and header in line:
+                value = line.strip().split("--{}=".format(header))[-1]
+                header_dict[header] = value
+                break
+    return header_dict
+
+
+def process_mem_per_cpu(mem_per_cpu):
+    """Convert a mem_per_cpu string to float"""
+    if isinstance(mem_per_cpu, str):
+        if mem_per_cpu[-1].upper() == "G":  # 0.5G to 0.5
+            mem_per_cpu = float(mem_per_cpu[:-1])  # 521MB to 0.5
+        elif mem_per_cpu[-2:] == "MB":
+            mem_per_cpu = float(mem_per_cpu[:-2]) / MB_PER_GB
+    else:
+        sys.exit("undefined mem per cpu, please add unit G/MB if not included")
+    return mem_per_cpu
+
+
+def process_header_values(slurm_header_dict, slurm_conf, hpc):
+    """
+       Process header dict values
+       For maui, mem_per_cpu is None as mem is not calculated against billing
+    """
+    if not slurm_header_dict[SlurmHeader.ntasks.value]:  # None, set to default
+        slurm_header_dict[SlurmHeader.ntasks.value] = DEFAULT_NTASKS
+    else:  # not None, convert str to int
+        slurm_header_dict[SlurmHeader.ntasks.value] = int(
+            slurm_header_dict[SlurmHeader.ntasks.value]
+        )
+
+    if not slurm_header_dict[SlurmHeader.cpus_per_task.value]:
+        slurm_header_dict[SlurmHeader.cpus_per_task.value] = DEFAULT_CPUS_PER_TASK
+    else:
+        slurm_header_dict[SlurmHeader.cpus_per_task.value] = int(
+            slurm_header_dict[SlurmHeader.cpus_per_task.value]
+        )
+
+    if hpc == HPC.mahuika.value:
```

Comment:
this if statement shouldnt be needed, if other functions are updated to not treat maui separately